app/controllers/developer_controller.rb

```diff
@@ -13,7 +13,18 @@ def update_sites
 
   def production_log
     @log = if params[:grep].present?
-             `tail -n 10000 log/production.log | grep -E '#{params[:grep]}' -C #{params[:context]} --color=never`
+             begin
+               grep_str = params[:grep].instance_of?(String) ? params[:grep] : params[:grep].to_s
+               context_int = params[:context].instance_of?(Integer) ? params[:context] : params[:context].to_i
+               # Check if coercion is successful, in case someone overloaded .to_*() methods
+               if !grep_str.instance_of?(String) || !context_int.instance_of?(Integer)
+                 raise ArgumentError, 'Coercion failure'
+               end
+             rescue
+               # Coercion failure -> deny access
+               raise ActionController::RoutingError, 'Not Found'
```

Comment:
This is the wrong error to raise, I'd probably `flash[:warning]` or something instead.